File: multi_agent/projects/im-desktop/src/services/user-profile-service.ts
/**
 * 用户资料服务
 * 处理用户资料、在线状态、好友备注等API调用
 */

import { ApiClient } from './api-client';
import {
  UserProfile,
  ProfileUpdateRequest,
  OnlineStatusRequest,
  OnlineStatus,
  FriendGroup,
  FriendRemark,
  FriendRemarkRequest
} from '../types/user-profile';

const api = ApiClient.getInstance();

class UserProfileService {
  
  /**
   * 获取当前用户资料
   */
  async getMyProfile(): Promise<UserProfile> {
    return api.get<UserProfile>('/api/v1/profile/me');
  }

  /**
   * 获取指定用户资料
   */
  async getProfile(userId: number): Promise<UserProfile> {
    return api.get<UserProfile>(`/api/v1/profile/${userId}`);
  }

  /**
   * 批量获取用户资料
   */
  async getProfiles(userIds: number[]): Promise<UserProfile[]> {
    return api.post<UserProfile[]>('/api/v1/profile/batch', { userIds });
  }

  /**
   * 更新个人资料
   */
  async updateProfile(request: ProfileUpdateRequest): Promise<UserProfile> {
    return api.put<UserProfile>('/api/v1/profile/me', request);
  }

  /**
   * 更新在线状态
   */
  async updateOnlineStatus(request: OnlineStatusRequest): Promise<UserProfile> {
    return api.put<UserProfile>('/api/v1/profile/me/status', request);
  }

  /**
   * 上传头像
   */
  async uploadAvatar(fileData: string): Promise<string> {
    const response = await api.post<{ avatarUrl: string }>('/api/v1/profile/me/avatar', { fileData });
    return response.avatarUrl;
  }

  /**
   * 获取好友分组列表
   */
  async getFriendGroups(): Promise<FriendGroup[]> {
    return api.get<FriendGroup[]>('/api/v1/profile/me/friend-groups');
  }

  /**
   * 创建好友分组
   */
  async createFriendGroup(groupName: string): Promise<FriendGroup> {
    return api.post<FriendGroup>('/api/v1/profile/me/friend-groups', { groupName });
  }

  /**
   * 更新好友备注
   */
  async updateFriendRemark(request: FriendRemarkRequest): Promise<FriendRemark> {
    return api.put<FriendRemark>(
      `/api/v1/profile/me/friend-remarks/${request.friendId}`,
      request
    );
  }

  /**
   * 获取好友备注
   */
  async getFriendRemark(friendId: number): Promise<FriendRemark | null> {
    try {
      return await api.get<FriendRemark>(`/api/v1/profile/me/friend-remarks/${friendId}`);
    } catch {
      return null;
    }
  }

  /**
   * 快速设置在线状态
   */
  async setOnlineStatus(status: OnlineStatus): Promise<void> {
    await this.updateOnlineStatus({ status });
  }

  /**
   * 设置自定义状态
   */
  async setCustomStatus(status: OnlineStatus, statusText: string): Promise<void> {
    await this.updateOnlineStatus({ status, statusText });
  }

  /**
   * 获取所有在线用户的ID列表
   */
  async getOnlineUsers(): Promise<number[]> {
    // 通过批量获取资料，检查哪些用户在线
    const profile = await this.getMyProfile();
    // 实际应通过WebSocket或专门的API获取
    return [profile.userId];
  }
}

export const userProfileService = new UserProfileService();

File: multi_agent/projects/im-mobile/lib/services/user_profile_service.dart
/**
 * 用户资料服务
 */

import 'package:im_mobile/models/user_profile.dart';
import 'package:im_mobile/services/api_client.dart';

class UserProfileService {
  final ApiClient _api = ApiClient();

  /**
   * 获取当前用户资料
   */
  Future<UserProfile> getMyProfile() async {
    final data = await _api.get('/api/v1/profile/me');
    return UserProfile.fromJson(data['data'] ?? data);
  }

  /**
   * 获取指定用户资料
   */
  Future<UserProfile> getProfile(int userId) async {
    final data = await _api.get('/api/v1/profile/$userId');
    return UserProfile.fromJson(data['data'] ?? data);
  }

  /**
   * 批量获取用户资料
   */
  Future<List<UserProfile>> getProfiles(List<int> userIds) async {
    final data = await _api.post('/api/v1/profile/batch', {'userIds': userIds});
    final list = data['data'] ?? data;
    return (list as List).map((e) => UserProfile.fromJson(e)).toList();
  }

  /**
   * 更新个人资料
   */
  Future<UserProfile> updateProfile(Map<String, dynamic> updates) async {
    final data = await _api.put('/api/v1/profile/me', updates);
    return UserProfile.fromJson(data['data'] ?? data);
  }

  /**
   * 更新在线状态
   */
  Future<UserProfile> updateOnlineStatus(OnlineStatus status, {String? statusText}) async {
    final data = await _api.put('/api/v1/profile/me/status', {
      'status': status.value,
      if (statusText != null) 'statusText': statusText,
    });
    return UserProfile.fromJson(data['data'] ?? data);
  }

  /**
   * 上传头像
   */
  Future<String> uploadAvatar(String base64Data) async {
    final data = await _api.post('/api/v1/profile/me/avatar', {
      'fileData': base64Data,
    });
    return data['data']['avatarUrl'] ?? '';
  }

  /**
   * 获取好友分组列表
   */
  Future<List<FriendGroup>> getFriendGroups() async {
    final data = await _api.get('/api/v1/profile/me/friend-groups');
    final list = data['data'] ?? data;
    return (list as List).map((e) => FriendGroup.fromJson(e)).toList();
  }

  /**
   * 创建好友分组
   */
  Future<FriendGroup> createFriendGroup(String groupName) async {
    final data = await _api.post('/api/v1/profile/me/friend-groups', {
      'groupName': groupName,
    });
    return FriendGroup.fromJson(data['data'] ?? data);
  }

  /**
   * 更新好友备注
   */
  Future<FriendRemark> updateFriendRemark(FriendRemark remark) async {
    final data = await _api.put(
      '/api/v1/profile/me/friend-remarks/${remark.friendId}',
      remark.toJson(),
    );
    return FriendRemark.fromJson(data['data'] ?? data);
  }

  /**
   * 获取好友备注
   */
  Future<FriendRemark?> getFriendRemark(int friendId) async {
    try {
      final data = await _api.get('/api/v1/profile/me/friend-remarks/$friendId');
      final remarkData = data['data'] ?? data;
      if (remarkData['id'] == null) return null;
      return FriendRemark.fromJson(remarkData);
    } catch (e) {
      return null;
    }
  }

  /**
   * 快速设置在线状态
   */
  Future<void> setOnlineStatus(OnlineStatus status) async {
    await updateOnlineStatus(status);
  }
}
